src/common/signal_handling.cpp

```diff
@@ -0,0 +1,32 @@
+#include "common/logging.h"
+#include "signal_handling.h"
+
+// We use signal() here instead of the usual strong recommendation for
+// using sigaction, which apparently is not available for Windows (cf.
+// https://stackoverflow.com/questions/231912/what-is-the-difference-between-sigaction-and-signal).
+
+namespace marian{
+volatile std::sig_atomic_t sigflags_{0};
+volatile std::sig_atomic_t gracefulExitRequested_{0};
+
+bool getSignalFlag(const int sig) {
+  // sig_atomic_t has 32 bits. We don't accommodate signals beyond that.
+  ABORT_IF(sig >= 32, "Signal out of range (must be < 32, is {}).", sig);
+  return sigflags_ & (1<<sig);
+}
+
+void requestGracefulExit(int sig) {
+  setSignalFlag(sig);         // keep track of triggering signal
+  gracefulExitRequested_ = 1; // set flag to exit gracefully
```

Comment:
gracefulExitRequested() returns a bool, but here you set it to an integer. Please choose one (`bool` if possible)

Reply:
This relies on implicit type conversion from int types to bool. sig_atomic_t is specifically meant to be used in signal handlers. This is run-of-the-mill signal handling, see "Compliant Solution (Writing volatile sig_atomic_t)" here: https://wiki.sei.cmu.edu/confluence/display/c/SIG31-C.+Do+not+access+shared+objects+in+signal+handlers.